viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        movie_data = None
+
+    else:
+        movie_data = {
+            'title': title,
+            'genre': genre,
+            'rating': rating
+            }
+
+    return movie_data
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, MOVIE_TITLE_1):
+    for movies in user_data.values():
+        for movie in movies:
+            if movie["title"] == MOVIE_TITLE_1:
+                add_movie = movies.pop(movies.index(movie))
+                user_data["watched"].append(add_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    try:
+        total = 0
+        num_movies = 0
+
+        for movies in user_data.values():
+            for movie in movies:
+                num_movies += 1
+                total += movie['rating']
+        average = total / num_movies
+    except ZeroDivisionError:
+        average = 0.0
+
+    return average
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"]:
+        most_popular_genre = []
+        for movies in user_data.values():
+            for movie in movies:
+                most_popular_genre.append(movie['genre'])
+        user_data = max(set(most_popular_genre), key = most_popular_genre.count)
+    else:
+        user_data = None
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique_watched = []
+    friend_1_watched_movies = user_data["friends"][0]["watched"]
+    friend_2_watched_movies = user_data["friends"][1]["watched"]
+
+    for movie in user_data["watched"]:
+        if movie in friend_1_watched_movies or movie in friend_2_watched_movies:
+            continue
+        else:
+            unique_watched.append(movie)
+
+    return unique_watched
+    
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    
+    friend_1_watched_movies = user_data["friends"][0]["watched"]
+    friend_2_watched_movies = user_data["friends"][1]["watched"]

```

Comment:
The comments on scaling from the `get_unique_watched` function also apply here. How can we take this approach and make it more generic so we can handle any number of friends?